### PR TITLE
drivers: serial: stm32: add support for RS485 configuration

### PR DESCRIPTION
**Pull request by arbrauns**
The stm32 UART can output a "driver enable" signal on the RTS pin that allows controlling e.g. external RS-485 drivers.

This can already be configured through the devicetree using the `de-*`
properties, but not through `uart_configure()`. This commit enables the use
of `.flow_ctrl=UART_CFG_FLOW_CTRL_RS485`.

This is supported on all devices other than l1, f1, f2, and f4.

### CODE REVIEW

**Comment by erwango** [comment]
Would you mind checking https://github.com/zephyrproject-rtos/zephyr/pull/57524 ?

**Comment by legoabram** [comment]
> For runtime configuration through uart_configure(), the existing RS485 flow control configuration is used.

I hadn't noticed that UART_CFG_FLOW_CTRL_RS485 was a thing! Lemme try implementing some of your changes.

**Comment by arbrauns** [comment]
Rebased, now only contains changes related to `uart_configure()`, since #57524 already solved the devicetree stuff.

File: drivers/serial/uart_stm32.c [hunk 1]
```diff
@@ -388,14 +416,17 @@ static inline enum uart_config_data_bits uart_stm32_ll2cfg_databits(uint32_t db,
...
  * @param  fc: Zephyr hardware flow control option.
  * @retval LL_USART_HWCONTROL_RTS_CTS, or LL_USART_HWCONTROL_NONE.
  */
 static inline uint32_t uart_stm32_cfg2ll_hwctrl(enum uart_config_flow_control fc)
 {
 	if (fc == UART_CFG_FLOW_CTRL_RTS_CTS) {
 		return LL_USART_HWCONTROL_RTS_CTS;
+	} else if (fc == UART_CFG_FLOW_CTRL_RS485) {
+		/* Driver Enable is handled separately */
+		return LL_USART_HWCONTROL_NONE;
```
Comment (erwango) [inline drivers/serial/uart_stm32.c:429]: Since the function already returns `LL_USART_HWCONTROL_NONE`, a mere comment would be enough.

**Review by erwango — CHANGES_REQUESTED** [review]
One comment, otherwise LGTM.

**Review by erwango — APPROVED** [review]

**Review by gautierg-st — APPROVED** [review]

**Pull request merged**